plutus-core/plutus-ir/test/prettyprintingReadable/listMatch.golden

```diff
@@ -0,0 +1,8 @@
+letrec data (List :: * -> *) a | match_List where
+         Nil : List a
+         Cons : a -> List a -> List a
+in match_List
+     {all (a :: *). a -> a} (Nil {all (a :: *). a -> a}) {all (a :: *). a -> a}
```

Comment:
"readable" :laughing: To be fair, this one is basically impossible to print "nicely".

Reply:
We had a version that would print this with unicode foralls and wouldn't print `*`. That's a lot more readable but in this version I wanted to make sure we at least try to remain somewhat uniform with the other prettyprinters.

Reply:
What do you think about the current version of how we pretty-print type binders here? (i.e. not showing `*`). I would like to make this the default for types as well, but then we'll have to do some work elsewhere too.

Reply:
`PlutusCore.Pretty.Readable` has 

```haskell
data ShowKinds
    = ShowKindsYes
    | ShowKindsNo
```

which determines whether to show kinds or not. We use it here:

```haskell
-- | Pretty-print a binding at the type level.
typeBinderDocM
    :: ( MonadReader env m, HasPrettyConfigReadable env configName
       , PrettyReadableBy configName tyname
       )
    => ((tyname -> Kind a -> Doc ann) -> AnyToDoc (PrettyConfigReadable configName) ann -> Doc ann)
    -> m (Doc ann)
typeBinderDocM k = do
    showKinds <- view $ prettyConfig . pcrShowKinds
    withPrettyAt ToTheRight botFixity $ \prettyBot -> do
        let prettyBind name kind = case showKinds of
                ShowKindsYes -> parens $ prettyBot name <+> "::" <+> prettyBot kind
                ShowKindsNo  -> prettyBot name
        encloseM binderFixity $ k prettyBind prettyBot
```

which omits kind annotations in binders when `ShowKindsNo` is chosen.

Do you think you could use this existing machinery?

Reply:
`typeBinderDocM` is not exported from `PlutusCore.Core.Instance.Pretty.Readable`, but do feel free to export and use it, if appropriate.

Reply:
Aha, just saw that you've already done that.

Reply:
Yeah, I'm confused by your comment Max, it looks like you're already using the printer that's sensitive to printing kinds, and also in this example the kinds _are_ printed?